tests/Gateway/Request/AuthenticateSPRequestTest.php

```diff
@@ -218,4 +213,11 @@ public function testNZDrivingLicence()
         $this->assertSame(123, $test->NewZealand->DrivingLicence->Version);
         $this->assertSame('ABC123', $test->NewZealand->DrivingLicence->VehicleRegistration);
     }
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->identity = new Identity();
+    }
```

Comment:
Is the `setUp` method supposed to be in the end of a test file by any guideline? I used to see it in the top.

Reply:
The only reason I moved it was because the `slevomat/coding-standard` comes with the ﻿[SlevomatCodingStandard.Classes.ClassStructure](https://github.com/slevomat/coding-standard#slevomatcodingstandardclassesclassstructure-) sniff which by default requires `public` methods to be above `protected` methods, and `protected` above `private`.

(Note: This is also why some other methods were moved around on other classes)

I like having `setUp` at the top too as it describes what happens before every test, but I didn't want to disable that sniff as it's generally a good idea to separate out public/protected/private methods, and you can't disable that sniff for this specific use case unfortunately.

My feeling is that it's better to live with this minor issue than remove the sniff, especially given `setUp` isn't actually doing anything ground-breaking or crazy here.

Reply:
Fair points.